results/cmd/api/main_test.go

```diff
@@ -437,3 +439,146 @@ func TestDeleteTaskRun(t *testing.T) {
 		t.Fatalf("same taskrun not supposed to be deleted again: %v", r.String())
 	}
 }
+func TestListTaskRuns(t *testing.T) {
+	// Create a temporary database
+	srv, err := setupTestDB(t)
+	if err != nil {
+		t.Fatalf("failed to setup db: %v", err)
+	}
+
+	// Create a bunch of taskruns for testing
+	t1 := &pb.TaskRun{
+		ApiVersion: "v1beta1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "31415926",
+			Name:      "taskrun",
+			Namespace: "default",
+		},
+	}
+	t2 := &pb.TaskRun{
+		ApiVersion: "v1alpha1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "43245243",
+			Name:      "task",
+			Namespace: "default",
+		},
+	}
+	t3 := &pb.TaskRun{
+		ApiVersion: "v1beta1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "1234556",
+			Name:      "mytaskrun",
+			Namespace: "demo",
+		},
+	}
+	t4 := &pb.TaskRun{
+		ApiVersion: "v1beta1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "543535",
+			Name:      "newtaskrun",
+			Namespace: "demo",
+		},
+	}
+	t5 := &pb.TaskRun{
+		ApiVersion: "v1alpha1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "543535",
+			Name:      "newtaskrun",
+			Namespace: "official",
+		},
+	}
+	testList := []*pb.TaskRun{t1, t2, t3, t4, t5}
+	tt := []struct {
+		name   string
+		list   []*pb.TaskRun
+		filter string
+		expect []*pb.TaskRun
+	}{
+		{
+			name:   "test simple query",
+			list:   testList,
+			filter: `taskrun.api_version=="v1beta1"`,
+			expect: []*pb.TaskRun{t1, t3, t4},
+		},
+		{
+			name:   "test logical &&",
+			list:   testList,
+			filter: `taskrun.api_version=="v1beta1" && taskrun.metadata.namespace=="demo"`,
+			expect: []*pb.TaskRun{t3, t4},
+		},
+		{
+			name:   "test logical ||",
```

Comment:
we may be able to combine some of these tests e.g. test a complex query that includes an logical AND, OR as well as startWith?

Reply:
I have thought for a while and decide to delete some of these tests, basically they are the same as a feature of cel. So a simple query test is enough

Reply:
done